### PR TITLE
Wii backport

### PR DESCRIPTION
**Pull request by palxex**
Leaving problems:

1. ~~save loading no problem but saving crash. Not sure why. From debugger it happend in PAL_SaveGame_Common DO_BYTESWAP(s, size), but when loading there's another BIG DO_BYTESWAP no problem, not sure whether I've made some stupid error. And then, anybody have idea on rapidly converting large memory block endian? So It seems a compiler error.~~ Finally it turned out to be a silly stupid error...
2. AVIPlay has more endian problem other than what I fixed in 81cb91f. Now 1.avi plays well, 2.avi will not play due to chunk reading wrong, 3.avi will play audio but no video. @CecilHarvey please look at it.

After all, all our main maintaining platforms is little endian now( iOS/android commonly uses little endian arm), its not a good thing for keep big-endian code remains healthily. I suggest picks up a big-endian platform to travis-ci.

### CODE REVIEW

**Comment by CecilHarvey** [comment]
Maybe I can debug big-endian issues locally with Mac OS X 10.4 (which supports Rosetta) under VMWare.

File: adplug/fprovide.cpp [hunk 1]
```diff
@@ -20,6 +20,9 @@
  */
 
 #include <string.h>
+#ifdef GEKKO
+#include <strings.h>
```
Comment (louyihua) [inline adplug/fprovide.cpp]: Better to have here include `pal_config.h` and include `strings.h` inside the configuration header.

File: adplug/rix.cpp [hunk 2]
```diff
@@ -213,27 +225,27 @@ float CrixPlayer::getrefresh()
 }
 
 /*------------------Implemention----------------------------*/
-inline void CrixPlayer::set_new_int()
```
Comment (palxex) [inline adplug/rix.cpp]: Oh, in fact its removed for ease of debugging, forgot restoring them after debugging. A macro is a good suggestion in debug purpose.

**Comment by palxex** [comment]
Yesterday I've tried upgrade to latest devkitpro(previous I'm using the toolchain downloaded about 2015), almost everything was mess up(although I've recompiled everything have ABI problem): RIX not work, draw font crash, not to say saving/loading, even debugging( either real wii wifi-debugging or dolphin virtual gecko COM-debugging ). I've to say this homebrew toolchain is too unstable to work with. Have to stay on old version( which only have the bug of cannot saving ), which I've written in travis file.

**Comment by CecilHarvey** [comment]
2.avi = no problems under Mac OS X 10.4 Rosetta.
btw, is this wii port still based on SDL1?

**Comment by palxex** [comment]
Yes.
If so...maybe also toolchain problem, let me try figure it whether could be fixed, or just left it like saving.

**Comment by palxex** [comment]
@CecilHarvey made a new commit that fixes the chunk reading problem on Wii which deny 2.avi from playing, please test whether it breaks Rosetta. From code it seems all big-endian systems should be affected.
For 3.avi still didn't find any clue why it don't show anything; maybe need leave it during this PR. All other avi plays well.

**Pull request merged**